simulators/ethereum/engine/test/spec.go

```diff
@@ -108,15 +108,27 @@ func (s Spec) GetForkConfig() ForkConfig {
 	return s.ForkConfig
 }
 
+func GenesisFactory(clientName string) helper.Genesis {
+	switch clientName {
+	case "erigon_":
+		return &helper.ErigonGenesis{}
+	case "nethermind_":
+		return &helper.NethermindChainSpec{}
+	default:
+		return nil
```

Comment:
nit: I guess it make sense to panic or return error here as upcoming `genesis.UnmarshalJSON` will cause nil pointer exception anyways, does it ?

Reply:
It will break in the next step, that's ok for now

Reply:
Yes and I meant that we can panic here to be sure where is the problem exactly (i.e. `unsupported client provided`)

Reply:
Done